File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_map = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in nums_map:
                return [i, nums_map[complement]]
            nums_map[nums[i]] = i
            
        return [0, 0]

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> nMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (nMap.containsKey(complement)) {
                int ans[] = {nMap.get(complement), i};
                return ans;
            }
            nMap.put(nums[i], i);
        }
        int ans[] = {0, 0};
        return ans;
    }
}